torch/_meta_registrations.py

```diff
@@ -6083,6 +6084,31 @@ def meta_bucketize(self, boundaries, *, out_int32=False, right=False):
     ).contiguous()
 
 
+@register_meta([aten.histc])
+@out_wrapper()
+def meta_histc(input, bins=100, min=0, max=0):
+    fn_name = "histc()"
+    torch._check(
+        input.is_floating_point(),
+        lambda: f"{fn_name}: input tensor must have a float dtype, not {input.dtype}",
```

Comment:
This limitation doesn't seem to exist in eager:
https://github.com/pytorch/pytorch/blob/277ab8a4c0d483bc0166a7eaa482a57a5c42c381/aten/src/ATen/native/cuda/SummaryOps.cu#L395

Reply:
oh you're right, I only checked this for `device="cpu"`, where it isn't implemented for integers. I'll restrict this check to `fake_device="cpu"`.